File: module_video/src/main/java/com/yyxnb/module_video/ui/find/VideoFindFragment.java
package com.yyxnb.module_video.ui.find;

import android.graphics.Color;
import android.os.Bundle;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.yyxnb.lib_adapter.BaseViewHolder;
import com.yyxnb.lib_adapter.ItemDecoration;
import com.yyxnb.lib_adapter.SimpleOnItemClickListener;
import com.yyxnb.lib_arch.annotations.BindRes;
import com.yyxnb.common_base.base.BaseFragment;
import com.yyxnb.module_video.R;
import com.yyxnb.module_video.adapter.VideoFindAdapter;
import com.yyxnb.module_video.config.DataConfig;
import com.yyxnb.module_video.databinding.FragmentVideoFindBinding;

/**
 * 发现
 */
@BindRes(subPage = true)
public class VideoFindFragment extends BaseFragment {

    private VideoFindAdapter mAdapter;

    private FragmentVideoFindBinding binding;
    private RecyclerView mRecyclerView;

    @Override
    public int initLayoutResId() {
        return R.layout.fragment_video_find;
    }

    @Override
    public void initView(Bundle savedInstanceState) {
        binding = getBinding();
        mRecyclerView = binding.rvContent;
        mAdapter = new VideoFindAdapter();
        mRecyclerView.setLayoutManager(new GridLayoutManager(getContext(), 2));
        ItemDecoration decoration = new ItemDecoration(getContext());
        decoration.setDividerColor(Color.WHITE)
                .setDividerHeight(2)
                .setDividerWidth(2)
                .setDrawBorderTopAndBottom(true)
                .setDrawBorderLeftAndRight(true);
        mRecyclerView.addItemDecoration(decoration);
        mRecyclerView.setHasFixedSize(true);
        mRecyclerView.setAdapter(mAdapter);
    }

    @Override
    public void initViewData() {
        mAdapter.setDataItems(DataConfig.getTikTokBeans());

        mAdapter.setOnItemClickListener(new SimpleOnItemClickListener() {
            @Override
            public void onItemClick(View view, BaseViewHolder holder, int position) {
                super.onItemClick(view, holder, position);
//                startFragment(VideoPlayFragment.newInstance(position, mAdapter.getData()));
            }
        });
    }

    @Override
    public void onVisible() {
        log("find v");
    }

    @Override
    public void onInVisible() {
        log("find iv");
    }

}

File: module_video/src/main/java/com/yyxnb/module_video/ui/find/VideoFindFragment.kt
package com.yyxnb.module_video.ui.find

import android.graphics.Color
import android.os.Bundle
import android.view.View
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.yyxnb.adapter.BaseViewHolder
import com.yyxnb.adapter.ItemDecoration
import com.yyxnb.adapter.MultiItemTypeAdapter
import com.yyxnb.adapter.SimpleOnItemClickListener
import com.yyxnb.arch.annotations.BindRes
import com.yyxnb.common.utils.log.LogUtils.w
import com.yyxnb.common_base.base.BaseFragment
import com.yyxnb.module_video.R
import com.yyxnb.module_video.adapter.VideoFindAdapter
import com.yyxnb.module_video.config.DataConfig.tikTokBeans
import com.yyxnb.module_video.databinding.FragmentVideoFindBinding

/**
 * 发现
 */
@BindRes(subPage = true)
class VideoFindFragment : BaseFragment() {

    private var mAdapter: VideoFindAdapter? = null
    private var binding: FragmentVideoFindBinding? = null
    private var mRecyclerView: RecyclerView? = null

    override fun initLayoutResId(): Int {
        return R.layout.fragment_video_find
    }

    override fun initView(savedInstanceState: Bundle?) {
        binding = getBinding()
        mRecyclerView = binding!!.mRecyclerView
        mAdapter = VideoFindAdapter()
        mRecyclerView!!.layoutManager = GridLayoutManager(context, 2)
        val decoration = ItemDecoration(context)
        decoration.setDividerColor(Color.WHITE)
                .setDividerHeight(2)
                .setDividerWidth(2)
                .setDrawBorderTopAndBottom(true).isDrawBorderLeftAndRight = true
        mRecyclerView!!.addItemDecoration(decoration)
        mRecyclerView!!.setHasFixedSize(true)
        mRecyclerView!!.adapter = mAdapter
    }

    override fun initViewData() {
        mAdapter!!.setDataItems(tikTokBeans)
        mAdapter!!.setOnItemClickListener(object : SimpleOnItemClickListener() {
            override fun onItemClick(view: View, holder: BaseViewHolder, position: Int) {
                super.onItemClick(view, holder, position)
                //                startFragment(VideoPlayFragment.newInstance(position, mAdapter.getData()));
            }
        })
    }

    override fun onVisible() {
        w("find v")
    }

    override fun onInVisible() {
        w("find iv")
    }
}